tests/test_shipment.py

```diff
@@ -244,3 +244,42 @@ def test_generate_form(one_call_buy_shipment, rma_form_options):
 
     assert form.form_type == form_type
     assert form.form_url is not None
+
+
+@pytest.mark.vcr()
+def test_shipment_create_with_carbon_offset(carbon_offset_shipment):
+    shipment = easypost.Shipment.create(carbon_offset=True, **carbon_offset_shipment)
+
+    assert isinstance(shipment, easypost.Shipment)
+    assert shipment.rates is not None
+    assert all(rate.carbon_offset is not None for rate in shipment.rates)
+
+
+@pytest.mark.vcr()
+def test_shipment_buy_with_carbon_offset(carbon_offset_shipment):
+    shipment = easypost.Shipment.create(**carbon_offset_shipment)
+    shipment.buy(carbon_offset=True, rate=shipment.lowest_rate())
+
+    assert isinstance(shipment, easypost.Shipment)
+    assert any(fee.type == "CarbonOffsetFee" for fee in shipment.fees)
+
+
+@pytest.mark.vcr()
+def test_shipment_one_call_buy_with_carbon_offset(carbon_offset_shipment_one_call_buy):
+    shipment = easypost.Shipment.create(carbon_offset=True, **carbon_offset_shipment_one_call_buy)
+
+    assert isinstance(shipment, easypost.Shipment)
+    assert all(rate.carbon_offset is not None for rate in shipment.rates)
+
+
+@pytest.mark.vcr()
+def test_shipment_rerate_with_carbon_offset(carbon_offset_shipment_one_call_buy):
+    shipment = easypost.Shipment.create(**carbon_offset_shipment_one_call_buy)
+
+    new_carbon_rates = shipment.regenerate_rates(carbon_offset=True)
+
+    assert all(rate.carbon_offset is not None for rate in new_carbon_rates.rates)
+
+    new_non_carbon_rates = shipment.regenerate_rates()
+    with pytest.raises(AttributeError):
```

Comment:
Why are we raising an AttributeError here?

Reply:
Because we are regenerating rate without `carbon_offset` and there will be no `carbon_offset` attribute in the `rate` object which will throw `AttributeError`, I don't think this test is necessary here because we have a specific unit test for `regenerate_rates()`, thought?

Reply:
I do a similar thing in the Ruby unit test, since, as Jack said, the `carbon_offset` property simply won't exist, so we need to catch the error of trying to access a non-existent property.

Please advise, whatever we do here we should do for Ruby (and I guess Node and PHP?)